File: src/Model/Obstacle.cpp
#include "Obstacle.h"

Obstacle::Obstacle(int xMin, int xMax, int yMin, int yMax, int zMin, int zMax)
	: m_xMin(xMin)
	, m_xMax(xMax)
	, m_yMin(yMin)
	, m_yMax(yMax)
	, m_zMin(zMin)
	, m_zMax(zMax)
{
}

int Obstacle::GetXMin() const { return m_xMin; }
int Obstacle::GetXMax() const { return m_xMax; }
int Obstacle::GetYMin() const { return m_yMin; }
int Obstacle::GetYMax() const { return m_yMax; }
int Obstacle::GetZMin() const { return m_zMin; }
int Obstacle::GetZMax() const { return m_zMax; }

File: src/Model/Obstacle.ts
export class Obstacle {
    private readonly xMin: number;
    private readonly xMax: number;
    private readonly yMin: number;
    private readonly yMax: number;
    private readonly zMin: number;
    private readonly zMax: number;

    constructor(xMin: number, xMax: number, yMin: number, yMax: number, zMin: number, zMax: number) {
        this.xMin = xMin;
        this.xMax = xMax;
        this.yMin = yMin;
        this.yMax = yMax;
        this.zMin = zMin;
        this.zMax = zMax;
    }

    getXMin(): number {
        return this.xMin;
    }

    getXMax(): number {
        return this.xMax;
    }

    getYMin(): number {
        return this.yMin;
    }

    getYMax(): number {
        return this.yMax;
    }

    getZMin(): number {
        return this.zMin;
    }

    getZMax(): number {
        return this.zMax;
    }
}
